dev/ToastNotifications/ToastActivationCallback.cpp

```diff
@@ -0,0 +1,45 @@
+﻿#include "pch.h"
+#include <ToastActivationCallback.h>
+#include "ToastActivatedEventArgs.h"
+#include "externs.h"
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/base.h>
+#include <winrt/Windows.ApplicationModel.Core.h>
+#include <iostream>
+#include <ToastNotificationUtility.h>
+
+namespace winrt
+{
+    using namespace Microsoft::Windows::ToastNotifications;
+    using namespace Windows::ApplicationModel::Core;
+    using namespace Windows::Foundation;
+    using namespace Windows::Foundation::Collections;
+}
+
+HRESULT __stdcall ToastActivationCallback::Activate(
+    LPCWSTR /* appUserModelId */,
+    LPCWSTR invokedArgs,
+    [[maybe_unused]] NOTIFICATION_USER_INPUT_DATA const* data,
+    [[maybe_unused]] ULONG dataCount)
+{
+    winrt::IMap<winrt::hstring, winrt::hstring> userInput{ winrt::single_threaded_map<winrt::hstring, winrt::hstring>() };
+    for (unsigned long i = 0; i < dataCount; i++)
+    {
+        userInput.Insert(data[i].Key, data[i].Value);
+    }
+
+    winrt::ToastActivatedEventArgs activatedEventArgs = winrt::make<winrt::Microsoft::Windows::ToastNotifications::implementation::ToastActivatedEventArgs>(invokedArgs, userInput);
+
+    if (GetToastHandleCount() > 0)
```

Comment:
Comment needed for the if else. It's unclear what each case is for. Per my understanding, one case is specifically for the Process Launch scenario while the other case is for the scenario where Process has already launched and you have registered event handlers.